app/src/main/java/org/wikipedia/analytics/TalkFunnel.kt

```diff
@@ -0,0 +1,54 @@
+package org.wikipedia.analytics
+
+import android.text.TextUtils
+import org.json.JSONObject
+import org.wikipedia.Constants
+import org.wikipedia.WikipediaApp
+import org.wikipedia.auth.AccountUtil.isLoggedIn
+import org.wikipedia.page.PageTitle
+import java.util.*
+
+class TalkFunnel constructor(private val title: PageTitle, private val invokeSource: Constants.InvokeSource) :
+        TimedFunnel(WikipediaApp.getInstance(), SCHEMA_NAME, REV_ID, SAMPLE_LOG_ALL) {
+
+    override fun preprocessData(eventData: JSONObject): JSONObject {
+        preprocessData(eventData, "source", invokeSource.getName())
+        preprocessData(eventData, "anon", !isLoggedIn)
+        preprocessData(eventData, "pageNS", if (!TextUtils.isEmpty(title.namespace))
```

Comment:
Should we use `!title.namespace.isNullOrEmpty()`?

Reply:
yep!